File: pascal'sTriangleI/pascalTriI.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        nums = [[0 for j in range(i+1)] for i in range(numRows)]
        if numRows<=0:return nums
        for i in range(0,numRows):
            for j in range(0,i+1):
                if j==0 or j==i: nums[i][j] = 1
                else: nums[i][j] = nums[i-1][j-1]+nums[i-1][j]
            i+=1
        return nums
            

File: pascal'sTriangleI/pascalTriI.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> nums = new ArrayList<List<Integer>>();
        if(numRows < 1) return nums;
        for(int i=0;i<numRows;i++){
            List<Integer> num = new ArrayList<Integer>();
            for(int j=0;j<i+1;j++){
                if(j==0 || j == i) num.add(1);
                else{
                   num.add(nums.get(i-1).get(j-1) + nums.get(i-1).get(j)); 
                } 
            }
            nums.add(num);
        }
        return nums;
    }
}